### ISSUE TITLE
`substring` fn not working correctly on `xt/id`

### ISSUE THREAD

**Issue by FiV0** [xtdb/crux#2768]
In the following test the second example (with `xt/id` fails)
```clj
(deftest substring-pred-not-working?
  (xt/submit-tx tu/*node* [[:put :xt-docs {:xt/id "foo-start" :v "foo-start"}]
                           [:put :xt-docs {:xt/id "bar-start" :v "bar-start"}]])

  (t/is (= [{:i "foo-start"}]
           (xt/q tu/*node*
                 '(-> (from :xt-docs [{:v i}])
                      (where (= "foo" (substring i 1 3))))))) 
  
  (t/is (= [{:i "foo-start"}]
           (xt/q tu/*node*
                 '(-> (from :xt-docs [{:xt/id i}])
                      (where (= "foo" (substring i 1 3))))))))
```

**Comment by FiV0** [comment]
Updated the test to XTQL.

**Comment by FiV0** [comment]
Fixed in 4c95c537
